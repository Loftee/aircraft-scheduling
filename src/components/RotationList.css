.rotation-list-wrapper {
    width: 37.5%;
    padding: 15px;
    float: left;
    list-style: none;
    margin: 0;
}

.rotation-list {
    width: 100%;
    float: left;
    list-style: none;
    padding: 0;
}

.rotation-list-item {
    width: 100%;
    margin-bottom: 20px;
    padding: 10px;
    border: 1px solid #ccc;
    cursor: pointer;
}

.rotation-list-item:hover {
    background: #eee;
    border-color: #999;
}

.rotation-list-item__data {
    font-size: 0;
}

.rotation-list-item__data span {
    font-size: initial;
    display: inline-block;
    width: 40%;
    vertical-align: middle;
    min-height: 20px;
}

.rotation-list-item__data span:last-child {
    text-align: right;
}

span.rotation-list-item__data-arrow {
    width: 20%;
    text-align: center;
    background: url('../assets/arrow.png') center center no-repeat;
    background-size: contain;

}

.rotation-list-reset {
    float: left;
    width: 100%;
    text-align: right;
}