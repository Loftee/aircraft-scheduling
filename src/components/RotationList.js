import React, {Component} from 'react';

import './RotationList.css';

class RotationList extends Component {
    render() {
        const {rotationList} = this.props;
        return (
            <div className={'rotation-list-wrapper'}>
                <h2>Rotation</h2>
                <ul className={'rotation-list'}>
                    {
                        rotationList.map((rotation, index) => {
                            return (
                                <li
                                    key={index}
                                    className={'rotation-list-item'}
                                >
                                    <h3 className={'rotation-list-item__header'}>Flight: {rotation.id}</h3>
                                    <div className={'rotation-list-item__data'}>
                                        <span>Departure: {rotation.readableDeparture} - {rotation.origin}</span>
                                        <span className={'rotation-list-item__data-arrow'}/>
                                        <span>Arrival: {rotation.readableArrival} - {rotation.destination}</span>
                                    </div>
                                </li>
                            )
                        })
                    }
                </ul>
            </div>
        )
    }
}

export default RotationList;