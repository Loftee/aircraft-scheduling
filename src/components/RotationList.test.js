import React from 'react';
import {shallow} from 'enzyme';
import RotationList from './RotationList';

describe('Rotation List', () => {
    const props = {
        rotationList:
            [{
                id: "AS1001",
                departureTime: 21600,
                arrivalTime: 26100,
                readableDeparture: "06:00",
                readableArrival: "07:15",
                origin: "LFSB",
                destination: "LFMN"
            }]
    };
    const component = shallow(<RotationList {...props} />);

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('displays selected rotations', () => {
       expect(component.find('.rotation-list-item').length).toBe(1);
    });
});