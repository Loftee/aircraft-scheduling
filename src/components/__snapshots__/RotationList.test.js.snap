// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rotation List renders correctly 1`] = `
<div
  className="rotation-list-wrapper"
>
  <h2>
    Rotation
  </h2>
  <ul
    className="rotation-list"
  >
    <li
      className="rotation-list-item"
      key="0"
    >
      <h3
        className="rotation-list-item__header"
      >
        Flight: 
        AS1001
      </h3>
      <div
        className="rotation-list-item__data"
      >
        <span>
          Departure: 
          06:00
           - 
          LFSB
        </span>
        <span
          className="rotation-list-item__data-arrow"
        />
        <span>
          Arrival: 
          07:15
           - 
          LFMN
        </span>
      </div>
    </li>
  </ul>
</div>
`;
